File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = 0;
        int minPrice = prices[0];
        for (int p : prices) {
            res = Math.max(res, p - minPrice);
            minPrice = Math.min(minPrice, p);
        }
        return res;

    }
}

File: Data Structures & Algorithms/merge-intervals/submission-0.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda p: p[0])  # 按照左端点从小到大排序

        res = []
        for p in intervals:
            if res and p[0] <= res[-1][1]: #左端点在合并区间内，可以合并
                res[-1][1] = max(res[-1][1], p[1])
            else:
                res.append(p)
        return res
